weasyl/controllers/user.py

```diff
@@ -42,6 +43,21 @@ def signin_post_(request):
         # Invalidate cached versions of the frontpage to respect the possibly changed SFW settings.
         index.template_fields.invalidate(logid)
         raise HTTPSeeOther(location=form.referer)
+    elif logid and logerror == "2fa":
+        # Password authentication passed, but user has 2FA set, so verify second factor (Also set SFW mode now)
+        if form.sfwmode == "sfw":
+            request.set_cookie_on_response("sfwmode", "sfw", 31536000)
+        index.template_fields.invalidate(logid)
+        # Store the authenticated userid & password auth time to the session
```

Comment:
When something that should be impossible happens, defaulting to disabling two-factor authentication and raising an error that isn’t reported via Sentry doesn’t seem like a good idea. Use a `ValueError` or something instead.